src/deadline/cinema4d_submitter/cinema4d_render_submitter.py

```diff
@@ -237,8 +237,42 @@ def _get_job_template(
     return job_template
 
 
+def _prompt_save_current_document():
+    doc = c4d.documents.GetActiveDocument()
+    if not doc.GetChanged():
+        # Document has no unsaved changed
+        return
+    if not c4d.gui.QuestionDialog("Save scene changes before submission?"):
+        # User selected No
+        return
+    file_path = doc.GetDocumentPath()
+    file_name = doc.GetDocumentName()
+    if file_path:
+        # Document save path exists
+        save_path = os.path.join(file_path, file_name)
+    else:
+        # Prompt with Save As to set path for Untitled document
+        save_path = c4d.storage.SaveDialog(c4d.FILESELECTTYPE_ANYTHING, "Save As", "c4d")
+        # Handle user cancels document save
+        if not save_path:
+            return
```

Comment:
If the program goes to line 258, does that mean the scene is not saved at all? Wouldn't that guarantee that the submission will not work? If so, maybe it would be better to fail fast than proceed to the submission dialog?

Reply:
Submission would work but render would fail. We can short circuit at this point.